docs/user/index.rst

```diff
@@ -0,0 +1,120 @@
+
+Each record in history has a date which is now a link to restore revision
+form.
+When you need to restore any object version you just need to click on desired
+date, check what will be changed  and either restore that revision or go back
+to object history.
+
+.. note::
+    Be aware that in current implementation revision is restored as is.
+    That means that any related object that is also stored in this revision
+    would be restored at the point at which that object was stored.
+    This is applicable to whole relation tree (if follow was configured).
+
+As soo as you restore object version the object would be at the version at
+which it was kept and you will be redirected to object edit form so that you
+can change object if you need to.
+
+You might also want to change related object (if applicable) since they also
+will be affected.
+
+.. note::
+   If :ref:`follow` was not configured properly and related object was not
+   stored in selected revision - you might end up with getting an
+   ``IntegrityError``. In this case you first need to restore related object
+   that was deleted.
+
+
+*************************
+Recovering deleted object
+*************************
+
+Second most powerful feature of ``aldryn-reversion`` is to recover deleted
+objects.
+You may find it on model's change list page after clicking on the
+``Recover deleted`` button on top of that page.
+
+If there are deleted objects (which do not exist in database but there is an
+object version stored with reversions) they will be listed on recover page.
+
+To recover deleted object simply choose an object, check the recover page
+instructions and if there is no conflicts on related fields (FKs) click on
+recover button with a label of ``Yes, I'm sure``.
+
+However if there are any conflicts that might be resolved by user (object
+model was registered with ``aldryn-reversions`` decorator AND a with admin
+mixin for model admin registration) you will see a list of links to recover
+related objects first.
+
+.. note::
+    In current implementation only required FKs are considered as a conflicts,
+    but this would be changed in next versions since if relation existed for
+    not required FK, and related object was deleted you will end up with
+    ``IntegrityError`` when you will try to restore this object.
+    In such cases if you know which related object is missing and if it was
+    registered with ``aldryn-reversion`` for revision tracking and with admin
+    mixin - you can recover related object first and then return to restore
+    this object.
+
+In case if related object was not registered with admin mixin but it was
+registered for revision tracking ``aldryn-reversion`` would try to solve
+conflict automatically by examining required FK relations.
+In case of success you might be able to restore the object and objects that
+are required relations to this object.
+
+.. note::
+    Be aware that automatic resolver would try to use selected object reivison
```

Comment:
"Note that automatic conflict resolution will try to use the selected object revision. Related objects will be restored to the version at which they were in that revision - not the object's latest version. Be sure to examine the related objects carefully and edit them so that they are in the desired state."